js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetsGroupedViewV2.tsx

```diff
@@ -0,0 +1,237 @@
+      .map(([key, {label, assets, link}], index) => {
+        const icon = (
+          <InsightsIcon
+            name={
+              selectedTab === 'kinds' && KNOWN_TAGS[label as KnownTagType]
+                ? (label as KnownTagType)
+                : propertyToLabelAndIcon[selectedTab].icon
+            }
+            size={16}
+          />
+        );
+        return (
+          <AssetSelectionSummaryListItem
+            key={key}
+            label={label}
+            icon={icon}
+            link={link}
+            assets={assets}
+            index={index}
+          />
+        );
+      });
+  },
+);
+
+const List = ({rows}: {rows: React.ReactNode[]}) => {
+  const scrollWrapperRef = useRef<HTMLDivElement | null>(null);
+  const rowVirtualizer = useVirtualizer({
+    count: rows.length,
+    getScrollElement: () => scrollWrapperRef.current,
+    estimateSize: () => 28,
+    overscan: 5,
+  });
+
+  const rowItems = rowVirtualizer.getVirtualItems();
+  const totalHeight = rowVirtualizer.getTotalSize();
+
+  return (
+    <Container ref={scrollWrapperRef} style={{maxHeight: '600px', overflowY: 'auto'}}>
+      <Inner $totalHeight={totalHeight}>
+        <div
+          style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            transform: `translateY(${rowItems[0]?.start ?? 0}px)`,
+          }}
+        >
+          {rowItems.map(({index, key}) => {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const row = rows[index]!;
+            return (
+              <div key={key} ref={rowVirtualizer.measureElement} data-index={index}>
+                {row}
+              </div>
+            );
+          })}
+        </div>
+      </Inner>
+    </Container>
+  );
+};
+
+const propertyToLabelAndIcon: Record<Property, {label: string; icon: IconName}> = {
+  groupName: {
+    label: 'Groups',
+    icon: 'asset_group',
+  },
+  kinds: {
+    label: 'Kinds',
+    icon: 'compute_kind',
+  },
+  owners: {
+    label: 'Owners',
+    icon: 'account_circle',
+  },
+  tags: {
+    label: 'Tags',
+    icon: 'tag',
+  },
+  repository: {
+    label: 'Code locations',
+    icon: 'folder',
+  },
+};
+
+// Removed accordion sections and create button; selections now render as a flat list
+
+const SelectionListItem = React.memo(
+  ({
+    index,
+    item,
+    assetsByAssetKey,
+  }: {
+    index: number;
+    item: ViewType;
+    assetsByAssetKey: Map<string, AssetTableFragment>;
+  }) => {
+    if (item.__typename === 'CatalogView') {
+      return <AssetSelectionSummaryListItemFromSelection index={index} item={item} />;
+    }
+    return (
+      <AssetSelectionSummaryListItem
+        index={index}
+        assets={item.assets
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          .map((assetKey: string) => assetsByAssetKey.get(assetKey)!)
+          .filter(Boolean)}
```

Comment:
You can avoid the non-null assertion by just allowing the `map` function to return null/undefined -- the subsequent `filter` call will eliminate the null/undefined values for you.

```
item.assets
  .map((assetKey: string) => assetsByAssetKey.get(assetKey))
  .filter(Boolean)
```